File: tests/bench/loop_i32.c
/* loop_i32.c — 与 tests/bench/loop_i32.su 对齐（LCG 混合，无 I/O） */
#include <stdint.h>

int main(void) {
  int32_t n = 100000000;
  int32_t s = 0;
  int32_t i = 0;
  while (i < n) {
    int32_t t = i * 1103515245 + 12345;
    s = s ^ t;
    i = i + 1;
    /* B-CMP：阻止 gcc/clang -O3 整循环常量折叠，与 shu_asm 真跑 1e8 次对齐。 */
    __asm__ volatile("" : "+r"(i), "+r"(s) : : "memory");
  }
  return (int)s;
}

File: tests/bench/loop_i32.zig
// loop_i32.zig — 与 tests/bench/loop_i32.sx / loop_i32.c 等价的 Zig 参照（run-perf-baseline.sh --bench）
const std = @import("std");

pub fn main() !void {
    const n: i32 = 1_000_000_000;
    var s: i32 = 0;
    var i: i32 = 0;
    while (i < n) : (i += 1) {
        s += 1;
    }
    try std.io.getStdOut().writer().print("{d}\n", .{s});
}
